lua/nvim-tree/actions/copy-paste.lua

```diff
@@ -13,39 +14,78 @@ local clipboard = {
 }
 
 local function do_copy(source, destination)
-  local source_stats = uv.fs_stat(source)
+  local source_stats, handle
+  local success, errmsg
 
-  if source_stats and source_stats.type == "file" then
-    return uv.fs_copyfile(source, destination)
+  source_stats, errmsg = uv.fs_stat(source)
+  if not source_stats then
+    log.line("copy_paste", "do_copy fs_stat '%s' failed '%s'", source, errmsg)
+    return false, errmsg
   end
 
-  local handle = uv.fs_scandir(source)
+  log.line("copy_paste", "do_copy %s '%s' -> '%s'", source_stats.type, source, destination)
 
-  if type(handle) == "string" then
-    return false, handle
+  if source == destination then
+    log.line("copy_paste", "do_copy source and destination are the same, exiting early")
+    return true
   end
 
-  uv.fs_mkdir(destination, source_stats.mode)
-
-  while true do
-    local name, _ = uv.fs_scandir_next(handle)
-    if not name then
-      break
+  if source_stats.type == "file" then
+    success, errmsg = uv.fs_copyfile(source, destination)
+    if not success then
+      log.line("copy_paste", "do_copy fs_copyfile failed '%s'", errmsg)
+      return false, errmsg
+    end
+    return true
+  elseif source_stats.type == "directory" then
+    handle, errmsg = uv.fs_scandir(source)
+    if type(handle) == "string" then
```

Comment:
I cannot imagine the situation in which a string is returned. I have left it here unchanged.